### PR TITLE
feat: add starter/BPMN for demo

### PR DESCRIPTION
**Pull request by npepinpe**
Here's the one PR to rule them all. This goes against all I preach: it's a huge PR, a complete thing all at once that cannot be reviewed :D

But this is just hacking right?

Anyway, I got it running in the Zeebe Cloud right now: https://grafana.dev.zeebe.io/goto/3luzmnkNg?orgId=1

[You can also explore the custom job worker metrics here](https://grafana.dev.zeebe.io/explore?panes=%7B%22gfo%22:%7B%22datasource%22:%22prometheus%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22sum%28zeebe_jobs_available_job%29%20by%20%28jobType%29%22,%22range%22:true,%22instant%22:true,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22prometheus%22%7D,%22editorMode%22:%22code%22,%22legendFormat%22:%22__auto%22%7D%5D,%22range%22:%7B%22from%22:%22now-15m%22,%22to%22:%22now%22%7D%7D%7D&schemaVersion=1&orgId=1).

The main downside though is that if your workers are fast (which they are babyyyyy), well, the available job count is always 0. So we'll need a double policy: scale workers up and down based on the perceived load from the workers. However, when scaled down to 0, we can use the custom job metric exporter to scale back up from 0. That's something we need to figure out how to do.

So I would ask you to become our HPA expert :heart:

### CODE REVIEW

File: demo/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Demo
+
+The demo will deploy the [sign-up-process.bpmn](./sign-up-process.bpmn) process, modeling a user sign up flow for the 
+premiere German cooking streaming service, Mettflix. This project provides deployment files to set up and operate the
```
Comment (Sijoma) [inline demo/README.md:4]: 😆 Haha. This is gold. Can we also do this in our presentation? Legend.

**Comment by npepinpe** [comment]
Since the PR is now gigantic, let's review it together.

**Comment by Sijoma** [comment]
> Since the PR is now gigantic, let's review it together.

Ahhh interesting. That's how you do it? I thought common practice is to "lgtm" in that case? :D

File: Makefile [hunk 2]
```diff
@@ -16,3 +16,32 @@ undeploy-camunda:
 
 teardown:
 	kind delete cluster --name hackdays
+
+### Demo stuff
+
+build-demo-starter:
+	docker build -t "gcr.io/zeebe-io/starter:autoscaling-demo" -f demo/starter.Dockerfile ./demo/
```
Comment (Sijoma) [inline Makefile:23]: Should we build these within this current repository? gcr.io/zeebe-io is private right?

File: demo/worker-payload.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"email":  "gunther.gunther@gunther.de"}
```
Comment (Sijoma) [inline demo/worker-payload.json:1]: My guy.

**Review by Sijoma — APPROVED** [review]
When did you write all the nice README information?

File: Makefile (hunk 2 above)
Comment (npepinpe) [reply Makefile:23]: It's private, yes. I can give you read access, many SREs have only read access, after all. I'm also fine just making you an honorary Zeebe dev (so you have some limited access to our GCP project, like other Zeebe devs).

File: demo/worker-payload.json (hunk 3 above)
Comment (npepinpe) [reply demo/worker-payload.json:1]: They call him Triple G

File: Makefile (hunk 2 above)
Comment (npepinpe) [reply Makefile:23]: We can push them here too, but last time it was sort of a pain to associate multiple containers to your repo, no? If you can figure out the easy way to do that, then sure :D

Comment (Sijoma) [reply Makefile:23]: > honorary Zeebe dev
I take that any day! 😄 Then we can skip that part for now and leave it like this. 

I think it shouldn't be too hard to push two images to the repo - I do so with our camunda-operator & kubectl-camunda-cli. But I don't want to waste time right now on figuring that out.

**Pull request merged**